React/Views/RCTBorderDrawing.m

```diff
@@ -498,6 +497,48 @@ static CGContextRef RCTUIGraphicsBeginImageContext(CGSize size, CGColorRef backg
   return image;
 }
 
+static id RCTGetBorderImageKey(RCTBorderStyle borderStyle,
+                               CGSize viewSize,
+                               RCTCornerRadii cornerRadii,
+                               UIEdgeInsets borderInsets,
+                               RCTBorderColors borderColors,
+                               CGColorRef backgroundColor,
+                               BOOL drawToEdge)
+{
+  // Important that this struct is aligned on all architectures, else
+  // the compiler can insert garbage into the padding and corrupt our hash.
+  struct {
+    RCTBorderStyle borderStyle;
+    CGSize viewSize;
+    RCTCornerRadii cornerRadii;
+    UIEdgeInsets borderInsets;
+    NSUInteger drawToEdge; // BOOL would be misaligned, not good.
+    CGFloat topColors[4];
+    CGFloat rightColors[4];
+    CGFloat bottomColors[4];
+    CGFloat leftColors[4];
+    CGFloat backgroundColors[4];
+  } borderParams = {
+    borderStyle,
+    viewSize,
+    cornerRadii,
+    borderInsets,
+    drawToEdge,
+    {0,0,0,0},
+    {0,0,0,0},
+    {0,0,0,0},
+    {0,0,0,0},
+    {0,0,0,0},
+  };
+  RCTGetRGBAColorComponents(borderColors.top, borderParams.topColors);
+  RCTGetRGBAColorComponents(borderColors.bottom, borderParams.bottomColors);
+  RCTGetRGBAColorComponents(borderColors.left, borderParams.leftColors);
+  RCTGetRGBAColorComponents(borderColors.right, borderParams.rightColors);
+  RCTGetRGBAColorComponents(backgroundColor, borderParams.backgroundColors);
+  
+  return [NSValue valueWithBytes:&borderParams objCType:@encode(typeof(borderParams))];
```

Comment:
You could use NSData here, and it would actually be slightly faster since the `isEqual:` wouldn't have to check the underlying type. But it also means another heap allocation – NSValue stores the data on the object itself instead of pointing into the heap. Plus it just feels more purpose-built.